micromanager_gui/main_window.py

```diff
@@ -220,16 +224,11 @@ def update_max_min(self, event=None):
 
         self.tab_wdg.max_min_val_label.setText(min_max_txt)
 
-    def snap(self):
+    def _snap(self, img: np.ndarray):
+        # update in a thread so we don't freeze UI
+        create_worker(self.update_viewer, img, _start_thread=True)
```

Comment:
```suggestion
    def _snap(self, img: np.ndarray):
        self.stop_live()
        # run snap  in a thread so we don't freeze UI
        create_worker(
            self._mmc.snap,
            _start_thread=True,
        )
```

This can't just update the viewer - it also needs to trigger a new snap so that it works for `snap_on_click`

Reply:
Can probably remove this entirely in the future once the new stage widget comes in

Reply:
@fdrgsp this still needs to be changed - currently snap on click is broken